clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -11460,6 +11460,15 @@ def err_sycl_expected_finalize_method : Error<
 def ext_sycl_2020_attr_spelling : ExtWarn<
   "use of attribute %0 is a SYCL 2020 extension">,
   InGroup<Sycl2017Compat>;
+def err_mixing_sycl_attr_csr_streaming : Error<
+  "at most one of ip_interface_csr and ip_interface_streaming can be applied to an IP component">;
+def err_mixing_sycl_attr_csr_stall : Error<
+  "ip_stall_free_return cannot be applied to an IP component with a CSR "
+  "interface. Please remove ip_stall_free_return or add ip_interface_streaming "
+  "to your IP component">;
+def err_sycl_attr_stall_requires_streaming_to_be_specified : Error<
+  "ip_stall_free_return requires ip_interface_streaming. Consider adding "
+  "ip_interface_streaming to your IP component, or removing ip_stall_free_return">;
```

Comment:
Why are we making this the user's problem instead of applying the `ip_interface_streaming` attribute for the user when they use `ip_stall_free_return`? (Also, the documentation is not clear that these are required to be in a pair.)